### PR TITLE
MM-31575 ENV var license override

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation for: https://mattermost.atlassian.net/browse/MM-31575

Updated:
- Self-Managed Admin Guide > Configure Mattermost > Configuration Settings
   - Changed the following bolded lines to proper headings: "Configuration in Database", "Environment Variables", and "Load Custom Configuration Defaults"
   - Added new subsection "Override Mattermost License File"

### CODE REVIEW

File: source/administration/config-settings.rst [hunk 1]
```diff
@@ -37,14 +39,26 @@ For any setting that is not set in ``config.json`` or in environment variables,
...
+--------------------------------
+
+Starting from Mattermost v5.26, you can specify a license override using an environment variable to override any license in the database or file configuration without replacing those licenses.
+
+When starting the server, specify the license key as ``MM_LICENSE`` with the contents of a license file.
+
+.. note::
+   If ``MM_LICENSE`` is set to a non-empty string, but the license specified is not valid, the Mattermost server will be started without a license.
+   
+   In a high-availability environment, using an environment variable to override a server license only affect the individual app server and doesn't propagate to other servers in the cluster.
```
Comment (wiggin77) [inline source/administration/config-settings.rst:52]: Minor typo...   'a server license only affect<ins>**s**</ins> the ...'

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:   with one minor typo.

File: source/administration/config-settings.rst (hunk 1 above)
Comment (cwarnermm) [reply source/administration/config-settings.rst]: Thanks! Fixed.

**Review by wiersgallak — APPROVED** [review]
LGTM

File: source/administration/config-settings.rst [hunk 2]
```diff
@@ -37,14 +39,26 @@ For any setting that is not set in ``config.json`` or in environment variables,
 .. warning::
    Database connection strings for the database read and search replicas need to be formatted using `URL encoding <https://www.w3schools.com/tags/ref_urlencode.asp>`__. Incorrectly formatted strings may cause some characters to terminate the string early, resulting in issues when the connection string is parsed.
    
-**Load Custom Configuration Defaults**
+Override Mattermost License File
+--------------------------------
+
+Starting from Mattermost v5.26, you can specify a license override using an environment variable to override any license in the database or file configuration without replacing those licenses.
```
Comment (justinegeffen) [inline source/administration/config-settings.rst:45]:
```suggestion
Starting from Mattermost v5.26, you can use an environment variable to override any license in the database or file configuration without replacing those licenses.
```

File: source/administration/config-settings.rst [hunk 3]
```diff
@@ -37,14 +39,26 @@ For any setting that is not set in ``config.json`` or in environment variables,
...
+
+When starting the server, specify the license key as ``MM_LICENSE`` with the contents of a license file.
+
+.. note::
+   If ``MM_LICENSE`` is set to a non-empty string, but the license specified is not valid, the Mattermost server will be started without a license.
+   
+   In a high-availability environment, using an environment variable to override a server license only affects the individual app server and doesn't propagate to other servers in the cluster.
```
Comment (justinegeffen) [inline source/administration/config-settings.rst:52]:
```suggestion
   In a High Availability deployment, using an environment variable to override a server license only affects the individual app server and doesn't propagate to other servers in the cluster.
```

**Review by justinegeffen — APPROVED** [review]

**Comment by justinegeffen** [comment]
@cwarnermm, just a reminder on this for my outstanding comments and your feedback. :)

**Pull request merged**